File: core/src/main/java/ru/pocketbyte/locolaser/utils/LogUtils.java
/*
 * Copyright © 2017 Denis Shurygin. All rights reserved.
 * Licensed under the Apache License, Version 2.0
 */

package ru.pocketbyte.locolaser.utils;

/**
 * @author Denis Shurygin
 */
public class LogUtils {

    public static void err(String message) {
        System.err.println("ERROR: " + message);
    }

    public static void err(Exception exception) {
        System.err.println("ERROR: ");
        exception.printStackTrace();
    }

    public static void warn(String message) {
        System.out.println("WARNING: " + message);
    }

    public static void info(String message) {
        System.out.println(message);
    }
}

File: core/src/main/java/ru/pocketbyte/locolaser/utils/LogUtils.kt
/*
 * Copyright © 2017 Denis Shurygin. All rights reserved.
 * Licensed under the Apache License, Version 2.0
 */

package ru.pocketbyte.locolaser.utils

/**
 * Utility object for logging errors, warnings, and informational messages.
 *
 * @author Denis Shurygin
 */
object LogUtils {

    /** Prints [message] to stderr prefixed with `"ERROR: "`. */
    fun err(message: String) {
        System.err.println("ERROR: $message")
    }

    /** Prints [exception] to stderr with its stack trace. */
    fun err(exception: Exception) {
        System.err.println("ERROR: ")
        exception.printStackTrace()
    }

    /** Prints [message] to stdout prefixed with `"WARNING: "`. */
    fun warn(message: String) {
        println("WARNING: $message")
    }

    /** Prints [message] to stdout. */
    fun info(message: String) {
        println(message)
    }
}
